launch/launch/conditions/launch_configuration_equals.py

```diff
@@ -36,13 +37,27 @@ class LaunchConfigurationEquals(Condition):
 
     If ``None`` is provided instead of a string expression, then the condition
     evaluates to ``True`` if the launch configuration is not set.
+
+    .. deprecated:: 1.1.0
+       Replaced by the more universally usable substitutions:
+       'EqualsSubstitution' and 'NotEqualsSubstitution'
     """
 
     def __init__(
         self,
         launch_configuration_name: Text,
         expected_value: Optional[SomeSubstitutionsType]
     ) -> None:
+        warnings.warn(
+            "The 'LaunchConfigurationEquals' and 'LaunchConfigurationNotEquals' Conditions are "
+            " deprecated. Use the 'EqualsSubstitution' and 'NotEqualsSubstitution' substitutions "
+            'instead! E.g.:\n'
+            '  IfCondition(\n  '
+            "\tEqualsSubstitution(LaunchConfiguration('some_launch_arg'), \"some_equality_check\")"
+            '\n  )',
+            UserWarning
+        )
+
```

Comment:
I wouldn't backport the deprecation notes/warnings

Reply:
Makes sense - was just a clean cherry-pick but I agree with making this modification. Done